public/app/core/utils/fetch.ts

```diff
@@ -105,7 +105,12 @@ export async function parseResponseBody<T>(
         return response.blob() as any;
 
       case 'json':
-        return response.json();
+        try {
+          return await response.json();
+        } catch (err) {
+          console.warn(`${response.url} returned an invalid JSON -`, err);
+          return {} as unknown as T;
```

Comment:
Is it only the empty body scenario we are trying to catch here? If that is the case it might be worth checking the `response.text()` so it is empty. If it isn't empty I would re-throw the error in the catch statement.

Reply:
I think that's a good point! Will change it accordingly 👍

Reply:
> unfortunately by that time the body stream is already read out, so we cannot read it out as a text again

oof, just spent a few hours with @bohandley re-discovering this rather unfortunate situation. this means that we cannot have downstream consumers attempt to parse the text without re-running the fetch request. :(

Reply:
https://stevenklambert.com/writing/fetch-json-text-fallback/

actually might still be do-able by using `response.clone()` a-priori, but that likely uses 2x the memory, which would suck for large responses.